### PR TITLE
Split shipping rates and shipping times section, and add automatic shipping rates option

### PR DESCRIPTION
**Pull request by ecgan**
### Changes proposed in this Pull Request:

Uses shipping rates API changes introduced in PR #1192.

Notable UI changes:

- Split the `CombinedShipping` UI into separate "Shipping rates" and "Shipping times" sections.
- Add "automatic" shipping rates option, and have it set as initial default value in Setup MC.

Other technical changes:

- Add "automatic" as a valid shipping rate value in `checkErrors` function.
- Add and fix tests for automatic shipping rate.
- Remove `CombinedShipping` from the code base.
    - Related `gla-shipping-title` CSS class is also removed and no longer in use in the code base.

Note: Changes to the shipping rate setup (i.e. minimum order amount for free shipping) will be done in a separate PR.

### References:

Figma: 2b5f7-pb

Internal P2 post: pcTzPl-ep-p2

### Screenshots:

Setup MC flow:

image

Edit free listings:

image

📹  Demo video with my voice annotation showing the Setup MC flow:

https://user-images.githubusercontent.com/417342/149969997-5fcb00bb-5c42-4173-8eba-5f29c0f6d374.mp4

### Detailed test instructions:

1. Go through Setup MC flow.
2. In Step 3 Configure your product listings: 
    - You should see separate shipping rates section and shipping times section.
    - The "automatic" shipping rate option should be selected by default. 
    - Shipping rate detailed setup should only be displayed when you select the "simple" option in shipping rate section.
    - Shipping time detailed setup should only be displayed when you select the "simple" option in shipping time section.
3. Complete the Setup MC flow. 
4. Go to the plugin's dashboard page, and edit the free listings program.
5. Go to Configure your product listings and edit the values there. It should behave similar to the Setup MC flow above.

### Changelog entry

> Add - Split shipping into separate shipping rates section and shipping times section, and add automatic option to shipping rates

### CODE REVIEW

File: js/src/components/free-listings/configure-product-listings/combined-shipping/index.scss
Comment (ecgan) [inline js/src/components/free-listings/configure-product-listings/combined-shipping/index.scss:1]: Checked that this is not used in the code base anymore.

File: js/src/components/free-listings/configure-product-listings/combined-shipping/index.js
Comment (ecgan) [inline js/src/components/free-listings/configure-product-listings/combined-shipping/index.js:30]: Checked that there is no more `CombinedShipping` in the code base anymore.

**Review by ecgan — COMMENTED** [review]
Added comments on some code checks that I have done.

File: js/src/components/free-listings/configure-product-listings/checkErrors.test.js
Comment (eason9487) [inline js/src/components/free-listings/configure-product-listings/checkErrors.test.js:76]: Haha, I never thought that one day the `'automatic'` would become a valid value. 😆

**Review by eason9487 — APPROVED** [review]
Tested locally. LGTM. A non-blocking item that could be adjusted is commented below.

💅 
The color of these texts in `<adioHelperText>` is different from the design, and their differences probably are meaningful.

image

image

File: js/src/components/free-listings/configure-product-listings/checkErrors.test.js
Comment (ecgan) [reply js/src/components/free-listings/configure-product-listings/checkErrors.test.js:76]: I checked the git log history and thought of asking you about it, but I figured you probably named it `'automatic'` because it is the opposite of the existing value `'manual'`. 😆

**Comment by ecgan** [comment]
@eason9487 ,

> The color of these texts in `<adioHelperText>` is different from the design, and their differences probably are meaningful.

Thanks. I added some changes (https://github.com/woocommerce/google-listings-and-ads/pull/1201/commits/d6a293a80285c7a41d28917fde369ea50f285c75) to make the `RadioHelperText` color match the design. The change affected the `"Can't find a country?"` text, so I made another change (https://github.com/woocommerce/google-listings-and-ads/pull/1201/commits/74fd000e064fa887b5daf58b851f76a747a4ec05) to make that text match the design.

Screenshot for the `RadioHelperText`:

image

Screenshot for the `"Can't find a country?"` text:

image

**Review by eason9487 — APPROVED** [review]
> I added some changes ([d6a293a](https://github.com/woocommerce/google-listings-and-ads/commit/d6a293a80285c7a41d28917fde369ea50f285c75)) to make the `RadioHelperText` color match the design. The change affected the `"Can't find a country?"` text, so I made another change ([74fd000](https://github.com/woocommerce/google-listings-and-ads/commit/74fd000e064fa887b5daf58b851f76a747a4ec05)) to make that text match the design.

Thanks! LGTM.

**Comment by ecgan** [comment]
Putting this PR on hold. I will merge this PR after the base PR #1192 got reviewed and approved.

**Comment by nima-karimi** [comment]
@ecgan I've merged #1192 into our feature branch. I think it should be okay to merge this one now 🙏

**Pull request merged**
